task.sql

```diff
@@ -1,17 +1,40 @@
 CREATE DATABASE ShopDB; 
 USE ShopDB; 
 
--- Create a table to store countries 
 CREATE TABLE Countries (
-    ID INT,
+    ID INT AUTO_INCREMENT,
     Name VARCHAR(50),
     PRIMARY KEY (ID)
 ) ENGINE=InnoDB;
 
--- Create a table for caching GeoIP data (Columns: ID, IP Range, CountryID)
+CREATE TABLE GeoIPCache (
+    ID INT AUTO_INCREMENT,
+    IPRange VARCHAR(50),
+    CountryID INT,
+    
+    PRIMARY KEY (ID),
+    FOREIGN KEY (CountryID) REFERENCES Countries (ID) ON DELETE CASCADE
+) ENGINE=Memory;
```

Comment:
The `GeoIPCache` table uses the `Memory` storage engine, which does not support foreign key constraints. Consider using a different storage engine like `InnoDB` if you need to enforce foreign key constraints.